### PR TITLE
[Integration][Jenkins] Add support for ingesting users

### PR DESCRIPTION
**Pull request by phalbert**
# Description

Add support for ingesting users from Jenkins server into Port. [PORT-6268 ](https://getport.atlassian.net/browse/PORT-6268)

## Type of change
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by PeyGis** [comment]
Looks good to me. One thing that is missing is bumping the integration version in pyproject.toml

File: integrations/jenkins/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 <!-- towncrier release notes start -->
 
+# 0.2.0 (2024-01-22)
+
+### Features
+
+- Added support for ingesting users (0.2.0)
```
Comment (PeyGis) [inline integrations/jenkins/CHANGELOG.md:14]: Remove the (0.2.0) Ideally, that number corresponds to the Jira ticket number

File: integrations/jenkins/client.py [hunk 2]
```diff
@@ -119,3 +119,29 @@ async def get_single_resource(self, resource_url: str) -> dict[str, Any]:
         )
         response.raise_for_status()
         return response.json()
+
+    async def get_users(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+        page_size = 5
```
Comment (PeyGis) [inline integrations/jenkins/client.py:124]: Any reason why the page size is very small?

**Review by PeyGis — CHANGES_REQUESTED** [review]

File: integrations/jenkins/client.py (hunk 2 above)
Comment (phalbert) [reply integrations/jenkins/client.py:124]: Testing. I should make that 50.

File: integrations/jenkins/pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "Jenkins"
-version = "0.1.0"
+version = "0.2.0"
```
Comment (PeyGis) [inline integrations/jenkins/pyproject.toml:3]: we increment it by 0.01 so this should be 0.1.1 the next improvement will be 0.1.2, 0.1.3 in that order

File: integrations/jenkins/CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+# 0.2.0 (2024-01-22)
+
+### Features
+
+- Added support for ingesting users (PORT-6268).
+
+### Improvements
+
+- Increase page size for jobs and build to 50.
```
Comment (PeyGis) [inline integrations/jenkins/CHANGELOG.md:19]: I don't think this is necessary. Remove it

**Review by PeyGis — CHANGES_REQUESTED** [review]
Minor changes

File: integrations/jenkins/client.py [hunk 5]
```diff
@@ -46,7 +48,7 @@ async def get_builds(self) -> AsyncGenerator[list[dict[str, Any]], None]:
     async def fetch_resources(
```
Comment (yairsimantov20) [inline integrations/jenkins/client.py:48]: You gave the other function a specific name.
i think this disserve one 2

File: integrations/jenkins/pyproject.toml (hunk 3 above)
Comment (phalbert) [reply integrations/jenkins/pyproject.toml:3]: @PeyGis  I would have, but I thought this was a new feature, so SemVer recommends incrementing the middle number.

File: integrations/jenkins/CHANGELOG.md (hunk 4 above)
Comment (phalbert) [reply integrations/jenkins/CHANGELOG.md:19]: Alright.

File: integrations/jenkins/pyproject.toml (hunk 3 above)
Comment (PeyGis) [reply integrations/jenkins/pyproject.toml:3]: Yeah I just read on the SemVer but let's stick with the ocean standard of 0.01. None of our integrations are at 0.2 yet

Comment (phalbert) [reply integrations/jenkins/pyproject.toml:3]: alright then.

**Review by Tankilevitch — APPROVED** [review]
LGTM

**Pull request merged**